Refreshing Fred Hutch WDL Content (#1276)
FredHutch/wiki: _datascience/using_workflows.md

- [WILDS WDL Library](/datascience/wilds_wdl/) - Ready-to-use modules and workflows
- [GATK Workflows](https://github.com/gatk-workflows)
- [PROOF](/datascience/proof/) - User-friendly workflow submission at Fred Hutch

### Nextflow

**Best for:**
- Users comfortable with Groovy/Java-like syntax
- Integration with [nf-core](https://nf-co.re/pipelines) community workflows
- Projects needing fine-grained control over execution

**Key features:**
- Mature ecosystem with extensive community support
- Large collection of pre-built workflows via [nf-core](https://nf-co.re/pipelines)
- Active Fred Hutch community

**Learn more:**
- [Nextflow at Fred Hutch](/compdemos/nextflow/) - Getting started guide
- [Nextflow Catalog](/datascience/nextflow_catalog) - Fred Hutch curated workflows
- [nf-core](https://nf-co.re/pipelines) - Community workflow catalog

## Choosing a Workflow System

Both WDL and Nextflow are excellent choices. Here's guidance on which might be better for your needs:

| Consideration | WDL | Nextflow |
|--------------|-----|----------|
| **Learning curve** | Gentle - simple, declarative syntax | Moderate - Groovy/Java-like syntax |

> | **Fred Hutch integration** | Excellent via PROOF | Excellent via direct execution |

Reviewer: Cirro?

| **Community resources** | Growing (WILDS, Broad, Terra) | Extensive (nf-core, global community) |
| **Execution options** | Multiple engines (Sprocket, miniWDL, Cromwell) | Nextflow runtime |
| **Local testing** | Easy with Sprocket/miniWDL | Easy with Nextflow |

> | **Pre-built workflows** | WILDS WDL Library, [GATK workflows](https://github.com/gatk-workflows) | nf-core (500+ workflows) |

Reviewer: BioWDL?

| **Best for** | All skill levels | Intermediate+ users |

### Decision Guide

**Choose WDL if you:**
- Are new to workflow systems

> - Want to use PROOF for easy cluster submission

Author: Absolutely, added in https://github.com/FredHutch/wiki/pull/1276/commits/40b00e5cec187e5b8bd794258782daec2ab6048c

### Execution Platforms

- **[PROOF](/datascience/proof/)** - User-friendly interface for WDL workflows on Fred Hutch cluster
- **[Gizmo](/scicomputing/compute_platforms/)** - Fred Hutch SLURM cluster for direct Nextflow execution
- **Local execution** - Sprocket, miniWDL, or Nextflow on your workstation
- **Cloud platforms** - AWS Batch, Terra, Cirro (via WDL or Nextflow)

## Training and Learning Resources

### WDL

- **[Data Science Lab: Developing WDL Workflows](https://hutchdatascience.org/Developing_WDL_Workflows/)** - Comprehensive course
- **[OpenWDL Learn WDL](https://github.com/openwdl/learn-wdl)** - Official tutorials
- **[WDL Documentation](https://wdl-docs.readthedocs.io/)** - Language reference

### Nextflow

- **[Nextflow Training](https://training.nextflow.io/)** - Official Nextflow training
- **[nf-core Usage Docs](https://nf-co.re/usage/introduction)** - Running nf-core pipelines
- **[Nextflow Patterns](https://nextflow-io.github.io/patterns/)** - Common workflow patterns

## Best Practices

Regardless of which workflow system you choose:

1. **Start small** - Test with small datasets before scaling up
2. **Use version control** - Track your workflows in Git

> 3. **Containerize** - Use Docker/Apptainer for reproducibility

Author: Good catch, added links for both the Docker and Apptainer articles in https://github.com/FredHutch/wiki/pull/1276/commits/40b00e5cec187e5b8bd794258782daec2ab6048c

- [PROOF](/datascience/proof/) platform for easy execution on Fred Hutch cluster

**Learn more:**
- [WDL Workflows Guide](/datascience/wdl_workflows/) - Language fundamentals
- [WDL Execution Engines](/datascience/wdl_execution_engines/) - How to run WDL workflows
- [WILDS WDL Library](/datascience/wilds_wdl/) - Ready-to-use modules and workflows
- [PROOF](/datascience/proof/) - User-friendly workflow submission at Fred Hutch

### Nextflow

**Best for:**
- Users comfortable with Groovy/Java-like syntax
- Workflows requiring complex logic or custom operations
- Integration with nf-core community workflows
- Projects needing fine-grained control over execution

**Key features:**
- Mature ecosystem with extensive community support
- Native support for containers, conda, and modules
- Powerful DSL (Domain Specific Language) for complex workflows
- Large collection of pre-built workflows via [nf-core](https://nf-co.re/pipelines)
- Active Fred Hutch community

**Learn more:**
- [Nextflow at Fred Hutch](/compdemos/nextflow/) - Getting started guide
- [Nextflow Catalog](/datascience/nextflow_catalog) - Fred Hutch curated workflows

> - [nf-core](https://nf-co.re/pipelines) - Community workflow catalog

Reviewer 2: ```suggestion
### WDL (Workflow Description Language)

**Best for:**
- Users who want a simple, human-readable workflow language
- Integration with Fred Hutch infrastructure via PROOF

**Key features:**
- Open-source language developed by the Broad Institute
- [WILDS WDL Library](/datascience/wilds_wdl/) provides tested, reusable WDLs built by and for Fred Hutch scientists
- [PROOF](/datascience/proof/) platform for easy execution on Fred Hutch cluster

**Learn more:**
- [WDL Workflows Guide](/datascience/wdl_workflows/) - Language fundamentals
- [WDL Execution Engines](/datascience/wdl_execution_engines/) - How to run WDL workflows
- [WILDS WDL Library](/datascience/wilds_wdl/) - Ready-to-use modules and workflows
- [PROOF](/datascience/proof/) - User-friendly workflow submission at Fred Hutch

### Nextflow

**Best for:**
- Users comfortable with Groovy/Java-like syntax
- Integration with [nf-core](https://nf-co.re/pipelines) community workflows
- Projects needing fine-grained control over execution

**Key features:**
- Mature ecosystem with extensive community support
- Native support for containers, conda, and modules
- Large collection of pre-built workflows via [nf-core](https://nf-co.re/pipelines)
- Active Fred Hutch community

**Learn more:**
- [Nextflow at Fred Hutch](/compdemos/nextflow/) - Getting started guide
- [Nextflow Catalog](/datascience/nextflow_catalog) - Fred Hutch curated workflows
- [nf-core](https://nf-co.re/pipelines) - Community workflow catalog
```

I would take out lines that are implied or don't help distinguish the two from each other. I would also try to note that with WDL we can take requests for WDL Library additions

## Choosing a Workflow System

Both WDL and Nextflow are excellent choices. Here's guidance on which might be better for your needs:

| Consideration | WDL | Nextflow |
|--------------|-----|----------|
| **Learning curve** | Gentle - simple, declarative syntax | Moderate - requires Groovy knowledge |
| **Fred Hutch integration** | Excellent via PROOF | Excellent via direct execution |
| **Community resources** | Growing (WILDS, Broad, Terra) | Extensive (nf-core, global community) |
| **Execution options** | Multiple engines (Sprocket, miniWDL, Cromwell) | Nextflow runtime |
| **Local testing** | Easy with Sprocket/miniWDL | Easy with Nextflow |
| **Pre-built workflows** | WILDS WDL Library, GATK workflows | nf-core (500+ workflows) |

> | **Language style** | Declarative (what to do) | Imperative (how to do it) |

Reviewer 2: ```suggestion
```
I don't think this line adds much to someones decision making

### Nextflow

- **[Nextflow Training](https://training.nextflow.io/)** - Official Nextflow training
- **[nf-core Usage Docs](https://nf-co.re/usage/introduction)** - Running nf-core pipelines
- **[Nextflow Patterns](https://nextflow-io.github.io/patterns/)** - Common workflow patterns

## Best Practices

Regardless of which workflow system you choose:

1. **Start small** - Test with small datasets before scaling up
2. **Use version control** - Track your workflows in Git
3. **Containerize** - Use Docker/Apptainer for reproducibility
4. **Document** - Include README files and example inputs
5. **Test** - Validate outputs before running large analyses
6. **Share** - Contribute back to community libraries
7. **Version workflows** - Tag releases when publishing results

## Advanced Topics

Once comfortable with basic workflows:

- **Cloud computing** - Scale to AWS Batch or other cloud platforms
- **Custom containers** - Build your own Docker images with [WILDS Docker Library](https://github.com/getwilds/wilds-docker-library)
- **Workflow optimization** - Fine-tune resource requests and parallelization
- **CI/CD** - Automate workflow testing with GitHub Actions

> - **Publishing** - Share workflows via GitHub for reproducible research

Reviewer 2: ```suggestion
```
I would cut this out since there is no guidance offered on these topics and people will come across these organically.

## Common Questions

**Q: Can I convert between WDL and Nextflow?**
A: Not directly. While both accomplish similar goals, their syntaxes are fundamentally different. Choose based on your needs and stick with it for a given project.

**Q: Which is faster?**
A: Performance depends more on your workflow design and resource allocation than the workflow system itself. Both are highly optimized.

**Q: Can I run WDL workflows without PROOF?**
A: Yes! You can use Sprocket, miniWDL, or Cromwell directly. See the [WDL Execution Engines guide](/datascience/wdl_execution_engines/).

**Q: Do I need to learn Docker?**

> A: Basic Docker knowledge is helpful but not required. Both WILDS WDL Library and nf-core provide pre-built containers. Workflow engines handle container pulling automatically.

Reviewer 2: ```suggestion
A: Basic Docker knowledge is very helpful. Pre-built Docker containers exist for both WDL and Nextflow and the workflow engines handle container pulling automatically. So while you may not need to build your own container, you are likely to be working with containers.
```
This has been everyone's biggest sticking point when getting started with WDL (including DaSL folks!) so I would just say yes to mentally prepare them
